File: C++/RichestCustomerWealth.cpp
/***********************************************************************************************
Problem      Richest Customer Wealth
Developer    Ian Hays
Date         1/30/2022
URL          https://leetcode.com/problems/richest-customer-wealth/
Space        O(1) 
Time         O(N^2)
Description  accumulate each account and return the maximum value. 
************************************************************************************************/

class Solution {
public:
    int maximumWealth(vector<vector<int>>& accounts) {
        int res = 0;
        for(auto& account: accounts){
            res = max(res, accumulate(begin(account), end(account),0));
        }
        return res;
    }
};

File: C#/RichestCustomerWealth.cs
//Ian Hays
//12-08-20
//https://leetcode.com/problems/richest-customer-wealth/
//SC: O(N^2) TC: O(N^2)

public class Solution {
    public int MaximumWealth(int[][] accounts)
    {
         return accounts.Select(row => row.Sum()).Max();    
    }
}
